cvmfs/catalog.h

```diff
@@ -231,6 +231,7 @@ class Catalog : public SingleCopy {
   Counters& GetCounters() { return counters_; }
 
   inline const CatalogDatabase &database() const { return *database_; }
+  inline       CatalogDatabase &database()       { return *database_; }
```

Comment:
Fine. However, I'm not sure what you mean. In a derived class `database()` is used to get an inherited field, either `mutable` or `const` depending on the calling method's const-ness. Or do you mean something else?